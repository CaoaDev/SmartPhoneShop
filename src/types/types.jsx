
export const types = {
    login: '[ aunth ] Login',
    logout: '[ aunth ] Logout',
}