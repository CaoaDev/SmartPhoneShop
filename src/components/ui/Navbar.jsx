import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { NavLink, useNavigate } from 'react-router-dom';
import { types } from '../../types/types';
import { Nav, Navbar } from 'react-bootstrap'

export const Navbar2 = () => {
    const { user, dispatch } = useContext( AuthContext );
    const navigate = useNavigate();
    const handleLogout = () => {
        // const action = {
        //     type: types.logout,
        //     payload: { logout: false }
        // }
        // dispatch( action );
        // navigate( '/Login', {
        //     replace: true
        // });
        dispatch( { type: types.logout } );
        navigate( '/Login', {
            replace: true
        });
    };

    return (
        // <nav className="navbar fixed-top navbar-expand-sm navbar-blue bg-dark">
        // <div className="navbar-collapse">
        // <div className="navbar-nav">
        <Navbar collapseOnSelect fixed='top' expand='sm' bg='dark' variant='dark' >
            <NavLink 
                className={ ( { isActive } ) => 'nav-item nav-link' + ( isActive ? ' active ' : '' ) }
                to=""
            >
                HÃ©roes
            </NavLink>
            <Navbar.Toggle aria-controls='respomsive-navbar-nav' />
            <Navbar.Collapse>
                <Nav>
                    <NavLink 
                        className={ ( { isActive } ) => 'nav-item nav-link' + ( isActive ? ' active ' : '' ) }
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        className={ ( { isActive } ) => 'nav-item nav-link' + ( isActive ? ' active ' : '' ) }
                        to="/dc"
                    >
                        DC
                    </NavLink>

                    <NavLink 
                        className={ ( { isActive } ) => 'nav-item nav-link' + ( isActive ? ' active ' : '' ) }
                        to="/search"
                    >
                        Search
                    </NavLink>
                </Nav>
            </Navbar.Collapse>
            <span className="text-info" >
                { user.name }
            </span>
            <button
                className="nav-item nav-link btn"
                onClick={ handleLogout }
                >
                    Logout
            </button>
        </Navbar>
    )
}