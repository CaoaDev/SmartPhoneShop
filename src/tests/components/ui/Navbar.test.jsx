import { MemoryRouter, Route, Routes } from "react-router-dom";
import { mount } from "enzyme";
import { Navbar2 } from "../../../components/ui/Navbar";
import { AuthContext } from "../../../auth/AuthContext";
import { types } from "../../../types/types";

const mockNavigate = jest.fn();

 jest.mock('react-router-dom', () => ({
     ...jest.requireActual('react-router-dom'),
     useNavigate: () => mockNavigate,
 }));

describe('Pruebas en <Navbar2 />', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Pedro',
            logged: true
        }
    }
        const wrapper = mount( // para hacer el mount de un componente tenemos q poner todos los q engloba
            <AuthContext.Provider value={ contextValue } >
                <MemoryRouter ImitialEntries={ [ '/' ] } > // poner el valor initial en este caso la pagina de raiz
                    <Routes >
                        <Route path='/' element={ <Navbar2 /> } />
                    </Routes>
                </MemoryRouter>
            </AuthContext.Provider>
        // pedro
        );
    test('Debe de mostrar correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( '.text-info' ).text().trim() ).toBe( 'Pedro' );
        // snapshot
        // .text-info = ... pedro
    });

    test('Debe de llamar el logout, llamar el navigate y el dispatch con los argumentos', () => {

        wrapper.find('button1').prop('onClick')();
        console.log(wrapper.html());

        expect( contextValue.dispatch ).toHaveBeenCalledWith({'type': types.logout });
        expect( mockNavigate ).toHaveBeenCalledWith('/Login', { replace: true });

    });
})