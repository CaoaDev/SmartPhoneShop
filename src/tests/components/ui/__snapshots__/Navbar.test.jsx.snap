// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar2 /> Debe de mostrar correctamente 1`] = `
Array [
  " // poner el valor initial en este caso la pagina de raiz",
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark fixed-top"
  >
    <a
      aria-current="page"
      className="nav-item nav-link active "
      href="/"
      onClick={[Function]}
    >
      HÃ©roes
    </a>
    <button
      aria-controls="respomsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <span
      className="text-info"
    >
      Pedro
    </span>
    <button1
      className="nav-item nav-link btn"
      onClick={[Function]}
    >
      Logout
    </button1>
  </nav>,
]
`;
